File: Programing_Basics_with_JS/conditionalStatementsMoreExercises/06.FlowerShop.js
function flowerShop(input) {
    let numberOfMagnolias = Number(input[0]);
    let numberOfHyacinths = Number(input[1]);
    let numberOfRoses = Number(input[2]);
    let numberOfCacti = Number(input[3]);
    let giftPrice = Number(input[4]);

    let magnoliaPrice = 3.25;
    let hyacinthPrice = 4.00;
    let rosePrice = 3.50;
    let cactusPrice = 8.00;

    let totalProfit = numberOfMagnolias * magnoliaPrice + numberOfHyacinths * hyacinthPrice + numberOfRoses * rosePrice + numberOfCacti * cactusPrice;
    let profitWithoutTaxes = totalProfit * 0.95;

    let moneyDiff = Math.abs(giftPrice - profitWithoutTaxes);

    if (profitWithoutTaxes >= giftPrice) {
        console.log(`She is left with ${Math.floor(moneyDiff)} leva.`);
    } else {
        console.log(`She will have to borrow ${Math.ceil(moneyDiff)} leva.`);
    }
}

flowerShop(['2', '3', '5', '1', '50'])
flowerShop(['15', '7', '5', '10', '100'])

File: Programming_Basics_with_Python/conditional_statements_more_exercise/06._flower_shop.py
from math import floor, ceil


def get_present(magnolias_count, hyacinths_count, roses_count, cacti_count, present_price, magnolias_price, hyacinths_price, roses_price, cacti_price, fee):
    total_profit = magnolias_count * magnolias_price + hyacinths_count * hyacinths_price + roses_count * roses_price + cacti_count * cacti_price
    total_profit_after_fee = total_profit - total_profit * fee
    money_difference = total_profit_after_fee - present_price

    if total_profit_after_fee >= present_price:
        return f'She is left with {floor(money_difference)} leva.'
    else:
        return f'She will have to borrow {ceil(abs(money_difference))} leva.'


magnolias_count = int(input())
hyacinths_count = int(input())
roses_count = int(input())
cacti_count = int(input())
present_price = float(input())

magnolias_price = 3.25
hyacinths_price = 4
roses_price = 3.50
cacti_price = 8
fee = 5 / 100

print(get_present(magnolias_count, hyacinths_count, roses_count, cacti_count, present_price, magnolias_price, hyacinths_price, roses_price, cacti_price, fee))